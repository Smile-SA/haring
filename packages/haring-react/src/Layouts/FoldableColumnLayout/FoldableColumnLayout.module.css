.box {
  padding: 40px 64px 0 64px;
  position: relative;
  width: 100%;

  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-md {
    display: flex;
    flex-direction: column;
    padding: 16px 24px;
  }
}

.boxWithMotif {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-md {
    padding: 24px;
  }
}

.collapseButton {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-md {
    padding: 0 24px;
    > button {
      width: 100%;

      > * {
        justify-content: space-between;
      }
    }
  }
  /*noinspection CssInvalidAtRule*/
  @mixin larger-than $mantine-breakpoint-md {
    display: none;
  }
}

.collapseButtonWithMotif {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-md {
    padding-top: 24px;
  }
}

.container {
  padding: 24px 64px 40px 64px;

  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-md {
    padding: 0 24px 24px 24px;
  }
}

.content {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-md {
    flex-basis: 100%;
    max-width: 100%;
    padding-top: 24px;
  }
}

.contentGrid {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-md {
    display: flex;
    flex-direction: column;
  }
}

.motif {
  left: -40px;
  pointer-events: none;
  position: absolute;
  top: -60px;
  user-select: none;
  z-index: 0;
}

.motifContainer {
  height: 100%;
  left: 0;
  overflow: hidden;
  position: absolute;
  top: 0;
  width: 100%;
}

.sidebar {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-md {
    flex-basis: 100%;
    max-width: 100%;
    padding-bottom: 0;
    padding-top: 8px;
  }
}

.topBlock {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-md {
    margin: 0;
  }
}

.topBlockContent {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-md {
    padding: 0;
  }
}

.topGrid {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-md {
    margin: 0;
  }
}

.topLeft {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-md {
    display: none;
  }
}

.topRight {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-md {
    flex-basis: 100%;
    max-width: 100%;
    padding: 0;
  }
}
