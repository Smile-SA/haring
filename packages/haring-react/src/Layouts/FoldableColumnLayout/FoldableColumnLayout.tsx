'use client';

import type {
  BoxProps,
  ContainerProps,
  MantineThemeOverride,
} from '@mantine/core';
import type { ReactElement, ReactNode } from 'react';

import { Button, Container, Grid, Paper, Switch, Text } from '@mantine/core';
import { useUncontrolled } from '@mantine/hooks';
import { CaretDown, CaretUp } from '@phosphor-icons/react';
import { NestedProvider, useMainTheme } from '@smile/haring-react-shared';

import classes from './FoldableColumnLayout.module.css';

export interface IFoldableColumnLayoutProps {
  boxMotif?: ReactNode;
  boxProps?: BoxProps;
  children: ReactNode;
  containerProps?: ContainerProps;
  defaultIsColumnVisible?: boolean;
  isColumnVisible?: boolean;
  onChangeIsColumnVisible?: (isVisible: boolean) => void;
  sidebarContent: ReactNode;
  sidebarToggleLabel?: string;
  topBarRight?: ReactNode;
  topBlock?: ReactNode;
  topBlockTheme?: MantineThemeOverride;
}

/**
 * Layout with header and grid content with foldable left sidebar
 */
export function FoldableColumnLayout(
  props: IFoldableColumnLayoutProps,
): ReactElement {
  const {
    boxMotif,
    boxProps,
    children,
    containerProps,
    defaultIsColumnVisible = true,
    isColumnVisible,
    onChangeIsColumnVisible,
    sidebarContent,
    sidebarToggleLabel = 'Display sidebar',
    topBarRight,
    topBlock,
    topBlockTheme,
  } = props;
  const [isColumnVisibleState, handleIsColumnVisibleChange] =
    useUncontrolled<boolean>({
      defaultValue: defaultIsColumnVisible,
      finalValue: true,
      onChange: onChangeIsColumnVisible,
      value: isColumnVisible,
    });
  const main = useMainTheme();
  const theme = topBlockTheme ?? main;

  function handleSidebarVisibleToggle(isVisible: boolean): void {
    handleIsColumnVisibleChange(isVisible);
  }

  return (
    <>
      <NestedProvider theme={theme}>
        <Paper
          className={`${classes.box} ${
            boxMotif ? classes.boxWithMotif : classes.boxWithoutMotif
          }`}
          {...boxProps}
        >
          {Boolean(boxMotif) && (
            <div className={classes.motifContainer}>
              <div className={classes.motif}>{boxMotif}</div>
            </div>
          )}
          {Boolean(topBlock) && (
            <Grid
              className={classes.topBlock}
              grow
              gutter={{ base: 0, md: 'lg' }}
            >
              <Grid.Col className={classes.topBlockContent} span={1}>
                {topBlock}
              </Grid.Col>
            </Grid>
          )}
          <Grid className={classes.topGrid} gutter={{ base: 0, md: 'lg' }}>
            <Grid.Col className={classes.topLeft} span={3}>
              <Switch
                checked={isColumnVisibleState}
                id="sidebar-toggle"
                label={
                  <Text fw={600} size="md">
                    {sidebarToggleLabel}
                  </Text>
                }
                onChange={(e) =>
                  handleSidebarVisibleToggle(Boolean(e.target.checked))
                }
              />
            </Grid.Col>
            {Boolean(topBarRight) && (
              <Grid.Col className={classes.topRight} span={9}>
                {topBarRight}
              </Grid.Col>
            )}
          </Grid>
        </Paper>
      </NestedProvider>
      <div
        className={`${classes.collapseButton} ${
          boxMotif ? classes.collapseButtonWithMotif : ''
        }`}
      >
        <Button
          onClick={() => handleSidebarVisibleToggle(!isColumnVisibleState)}
          rightSection={isColumnVisibleState ? <CaretUp /> : <CaretDown />}
        >
          {sidebarToggleLabel}
        </Button>
      </div>
      <Container className={classes.container} fluid {...containerProps}>
        <Grid
          className={classes.contentGrid}
          gutter="lg"
          overflow="hidden"
          pt={12}
        >
          <Grid.Col
            aria-hidden={!isColumnVisibleState}
            className={classes.sidebar}
            hidden={!isColumnVisibleState}
            span={3}
          >
            {sidebarContent}
          </Grid.Col>
          <Grid.Col
            className={classes.content}
            span={!isColumnVisibleState ? 12 : 9}
          >
            {children}
          </Grid.Col>
        </Grid>
      </Container>
    </>
  );
}
