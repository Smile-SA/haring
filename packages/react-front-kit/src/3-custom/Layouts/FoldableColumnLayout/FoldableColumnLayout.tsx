'use client';

import type { IHeaderProps } from '../../Components/Header/Header';
import type { ContainerProps } from '@mantine/core';
import type { AppShellProps } from '@mantine/core/lib/AppShell/AppShell';
import type { ChangeEvent, ReactElement, ReactNode } from 'react';

import { AppShell, Container, Grid, Switch, Text } from '@mantine/core';
import { useState } from 'react';

interface IFoldableColumnLayoutProps {
  appShellProps?: Omit<AppShellProps, 'children'>;
  children: ReactNode;
  containerProps?: ContainerProps;
  headerProps?: IHeaderProps;
  sidebarContent: ReactNode;
  sidebarToggleLabel?: string;
}

/**
 * Layout with header and grid content with foldable left sidebar
 */
export function FoldableColumnLayout(
  props: IFoldableColumnLayoutProps,
): ReactElement {
  const {
    appShellProps,
    children,
    containerProps,
    sidebarContent,
    sidebarToggleLabel = 'Display sidebar',
  } = props;

  const [isColumnVisible, setIsColumnVisible] = useState(true);

  function handleSidebarVisibleToggle(e: ChangeEvent<HTMLInputElement>): void {
    setIsColumnVisible(Boolean(e.target.checked));
  }

  return (
    <AppShell padding={0} {...appShellProps}>
      <Container fluid p="40px 64px" {...containerProps}>
        <Grid grow>
          <Grid.Col span={3}>
            <Switch
              checked={isColumnVisible}
              id="sidebar-toggle"
              label={
                <Text fw={600} size="md">
                  {sidebarToggleLabel}
                </Text>
              }
              onChange={handleSidebarVisibleToggle}
            />
          </Grid.Col>
        </Grid>
        <Grid grow gutter="xl">
          {Boolean(isColumnVisible) && (
            <Grid.Col span={3}>{sidebarContent}</Grid.Col>
          )}
          <Grid.Col span={9}>{children}</Grid.Col>
        </Grid>
      </Container>
    </AppShell>
  );
}
