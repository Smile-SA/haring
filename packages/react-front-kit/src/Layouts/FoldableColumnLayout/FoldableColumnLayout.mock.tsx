import type { ReactElement } from 'react';

import { primaryTheme } from '@smile/react-front-kit-shared';

import { Header } from '../../Components/Header/Header';
import { SidebarMenu } from '../../Components/SidebarMenu/SidebarMenu';
import { menuMock } from '../../Components/SidebarMenu/SidebarMenu.mock';

export const header: ReactElement = (
  <Header
    childrenComponent="nav"
    onSearchSubmit={(e) => {
      e.preventDefault();
    }}
    searchTheme={primaryTheme}
  >
    <span>Example Header...</span>
  </Header>
);

export const sidebarContent: ReactElement = (
  <>
    <span>Example Sidebar Content...</span>
    <SidebarMenu menu={menuMock} />
  </>
);

export const mainContent: ReactElement = (
  <>
    <span>Example Main content...</span>
    <p>
      Lorem <a href="#">ipsum</a> dolor sit amet, <b>consectetur</b> adipiscing
      elit. Sed varius <i>bibendum</i> dui non imperdiet. Donec vehicula
      fringilla lorem vitae rutrum. Etiam malesuada ullamcorper aliquam.
      Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere
      cubilia curae; Cras elit lacus, viverra vitae risus et, pharetra tincidunt
      felis. Aliquam erat volutpat. In vitae nibh eu turpis commodo luctus vitae
      id libero. Curabitur eget nunc volutpat, luctus quam rutrum, ultricies
      tellus. Integer diam nulla, vestibulum id enim quis, molestie luctus
      magna. Phasellus et rhoncus augue, id maximus mi. Vivamus consequat quam
      tristique ex laoreet, ut eleifend eros sodales. Cras bibendum enim dolor,
      id rutrum urna vestibulum non.
    </p>
    <p>
      Ut quis urna pharetra, elementum elit vel, venenatis sem. In tristique in
      ante venenatis consectetur. Sed laoreet pellentesque enim. Interdum et
      malesuada fames ac ante ipsum primis in faucibus. Mauris lobortis in diam
      non fringilla. Etiam maximus enim eget dolor tempus sodales. In sed turpis
      feugiat, sodales tortor vitae, semper odio. Donec vulputate erat orci, sit
      amet ultrices magna sodales sed. Curabitur tincidunt nunc id velit
      imperdiet, eget tempor mi interdum. Phasellus hendrerit sem nec venenatis
      suscipit.
    </p>
    <p>
      Suspendisse potenti. Fusce id nibh tempor mi interdum pretium vitae non
      purus. Pellentesque a nibh mattis, gravida diam id, fringilla nisi.
      Phasellus lacus magna, finibus et pharetra suscipit, accumsan in nisi.
      Curabitur sed tortor sodales, ornare tellus sed, egestas mauris. Fusce
      ipsum lectus, lacinia in tortor eu, consequat eleifend ipsum. Aenean
      ornare egestas ullamcorper. Nam metus justo, mollis non egestas at,
      tincidunt a enim. Phasellus vitae odio non metus accumsan dictum. Nunc
      posuere convallis dolor sed fringilla. Nullam ante felis, suscipit vitae
      iaculis ultricies, commodo non velit.
    </p>
  </>
);
