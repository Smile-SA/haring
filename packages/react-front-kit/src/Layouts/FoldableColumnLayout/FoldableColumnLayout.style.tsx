import { createStyles } from '@mantine/styles';

export const useStyles = createStyles((theme) => ({
  box: {
    padding: '40px 64px 0 64px',
    position: 'relative',
    [theme.fn.smallerThan('md')]: {
      display: 'flex',
      flexDirection: 'column',
      padding: '16px',
    },
    width: '100%',
  },
  boxWithMotif: {
    [theme.fn.smallerThan('md')]: {
      padding: '24px',
    },
  },
  collapseButton: {
    [theme.fn.smallerThan('md')]: {
      '> button': {
        '> *': {
          justifyContent: 'space-between',
        },
        width: '100%',
      },
      padding: '0 24px',
    },
    [theme.fn.largerThan('md')]: {
      display: 'none',
    },
  },
  collapseButtonWithMotif: {
    [theme.fn.smallerThan('md')]: {
      paddingTop: '24px',
    },
  },
  container: {
    padding: '24px 64px 40px 64px',
    [theme.fn.smallerThan('md')]: {
      padding: '0 24px 24px 24px',
    },
  },
  content: {
    [theme.fn.smallerThan('md')]: {
      flexBasis: '100%',
      maxWidth: '100%',
      paddingTop: 24,
    },
  },
  contentGrid: {
    [theme.fn.smallerThan('md')]: {
      display: 'flex',
      flexDirection: 'column',
    },
  },
  motif: {
    left: -40,
    pointerEvents: 'none',
    position: 'absolute',
    top: -60,
    userSelect: 'none',
    zIndex: 0,
  },
  motifContainer: {
    height: '100%',
    left: 0,
    overflow: 'hidden',
    position: 'absolute',
    top: 0,
    width: '100%',
  },
  sidebar: {
    [theme.fn.smallerThan('md')]: {
      flexBasis: '100%',
      maxWidth: '100%',
      paddingBottom: 0,
      paddingTop: 8,
    },
  },
  topBlock: {
    [theme.fn.smallerThan('md')]: {
      margin: 0,
    },
  },
  topBlockContent: {
    [theme.fn.smallerThan('md')]: {
      padding: 0,
    },
  },
  topGrid: {
    [theme.fn.smallerThan('md')]: {
      margin: 0,
    },
  },
  topLeft: {
    [theme.fn.smallerThan('md')]: {
      display: 'none',
    },
  },
  topRight: {
    [theme.fn.smallerThan('md')]: {
      flexBasis: '100%',
      maxWidth: '100%',
      padding: 0,
    },
  },
}));
