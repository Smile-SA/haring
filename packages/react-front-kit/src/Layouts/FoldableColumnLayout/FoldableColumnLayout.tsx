'use client';

import type {
  BoxProps,
  ContainerProps,
  MantineThemeOverride,
} from '@mantine/core';
import type { ChangeEvent, ReactElement, ReactNode } from 'react';

import {
  Box,
  Container,
  Grid,
  MantineProvider,
  Switch,
  Text,
} from '@mantine/core';
import { useUncontrolled } from '@mantine/hooks';
import { createStyles } from '@mantine/styles';
import { useMainTheme } from '@smile/react-front-kit-shared';

const useStyles = createStyles({
  box: {
    position: 'relative',
    width: '100%',
  },
  motif: {
    left: -40,
    position: 'absolute',
    top: -60,
    zIndex: 0,
  },
  motifContainer: {
    height: '100%',
    left: 0,
    overflow: 'hidden',
    position: 'absolute',
    top: 0,
    width: '100%',
  },
});

export interface IFoldableColumnLayoutProps {
  boxMotif?: ReactNode;
  boxProps?: BoxProps;
  children: ReactNode;
  containerProps?: ContainerProps;
  defaultIsColumnVisible?: boolean;
  isColumnVisible?: boolean;
  onChangeIsColumnVisible?: (isVisible: boolean) => void;
  sidebarContent: ReactNode;
  sidebarToggleLabel?: string;
  topBarRight?: ReactNode;
  topBlock?: ReactNode;
  topBlockTheme?: MantineThemeOverride;
}

/**
 * Layout with header and grid content with foldable left sidebar
 */
export function FoldableColumnLayout(
  props: IFoldableColumnLayoutProps,
): ReactElement {
  const {
    boxMotif,
    boxProps,
    children,
    containerProps,
    defaultIsColumnVisible = true,
    isColumnVisible,
    onChangeIsColumnVisible,
    sidebarContent,
    sidebarToggleLabel = 'Display sidebar',
    topBarRight,
    topBlock,
    topBlockTheme,
  } = props;
  const [isColumnVisibleState, handleIsColumnVisibleChange] =
    useUncontrolled<boolean>({
      defaultValue: defaultIsColumnVisible,
      finalValue: true,
      onChange: onChangeIsColumnVisible,
      value: isColumnVisible,
    });
  const main = useMainTheme();
  const theme = topBlockTheme ?? main;
  const { classes } = useStyles();

  function handleSidebarVisibleToggle(e: ChangeEvent<HTMLInputElement>): void {
    handleIsColumnVisibleChange(Boolean(e.target.checked));
  }

  return (
    <>
      <MantineProvider theme={theme}>
        <Box
          className={classes.box}
          color="primary"
          p="40px 64px 0 64px"
          {...boxProps}
        >
          {Boolean(boxMotif) && (
            <div className={classes.motifContainer}>
              <div className={classes.motif}>{boxMotif}</div>
            </div>
          )}
          {Boolean(topBlock) && (
            <Grid grow>
              <Grid.Col span={1}>{topBlock}</Grid.Col>
            </Grid>
          )}
          <Grid>
            <Grid.Col span={3}>
              <Switch
                checked={isColumnVisibleState}
                id="sidebar-toggle"
                label={
                  <Text fw={600} size="md">
                    {sidebarToggleLabel}
                  </Text>
                }
                onChange={handleSidebarVisibleToggle}
              />
            </Grid.Col>
            <Grid.Col span={9}>{topBarRight}</Grid.Col>
          </Grid>
        </Box>
      </MantineProvider>
      <Container fluid p="24px 64px 40px 64px" {...containerProps}>
        <Grid gutter="xl" pt={12}>
          <Grid.Col
            aria-hidden={!isColumnVisibleState}
            hidden={!isColumnVisibleState}
            span={3}
          >
            {sidebarContent}
          </Grid.Col>
          <Grid.Col span={!isColumnVisibleState ? 12 : 9}>{children}</Grid.Col>
        </Grid>
      </Container>
    </>
  );
}
