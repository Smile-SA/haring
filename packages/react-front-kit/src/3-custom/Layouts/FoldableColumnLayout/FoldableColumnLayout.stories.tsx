import type { Meta, StoryObj } from '@storybook/react';

import { useStorybookArgsConnect } from '../../../hooks/useStorybookArgsConnect';

import { FoldableColumnLayout as Cmp } from './FoldableColumnLayout';
import {
  header,
  mainContent,
  sidebarContent,
} from './FoldableColumnLayout.mock';

const meta = {
  component: Cmp,
  decorators: [
    function Component(Story, ctx) {
      const args = useStorybookArgsConnect(ctx.args, {
        onSearchChange: 'searchValue',
      });
      return <Story args={{ ...args }} />;
    },
  ],
  parameters: {
    layout: 'fullscreen',
  },
  title: '3-Custom/Layouts/FoldableColumnLayout',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const FoldableColumnLayout: IStory = {
  args: {
    appShellProps: {
      header,
      padding: 0,
    },
    children: mainContent,
    containerProps: { p: '20px' },
    sidebarContent,
    sidebarToggleLabel: 'View sidebar',
  },
};
