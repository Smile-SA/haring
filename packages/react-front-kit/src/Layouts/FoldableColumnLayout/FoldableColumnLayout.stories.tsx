import type { Meta, StoryObj } from '@storybook/react';

import { useStorybookArgsConnect } from '@smile/react-front-kit-shared/storybook-utils';

import { FoldableColumnLayout as Cmp } from './FoldableColumnLayout';
import { mainContent, sidebarContent } from './FoldableColumnLayout.mock';

const meta = {
  component: Cmp,
  decorators: [
    function Component(Story, ctx) {
      const args = useStorybookArgsConnect(ctx.args, {
        onSearchChange: 'searchValue',
      });
      return <Story args={{ ...args }} />;
    },
  ],
  parameters: {
    layout: 'fullscreen',
  },
  title: '3-Custom/Layouts/FoldableColumnLayout',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const FoldableColumnLayout: IStory = {
  args: {
    children: mainContent,
    containerProps: { p: '20px' },
    sidebarContent,
    sidebarToggleLabel: 'View sidebar',
  },
};
