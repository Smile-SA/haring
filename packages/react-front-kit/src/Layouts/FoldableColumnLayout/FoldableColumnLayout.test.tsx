import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { FoldableColumnLayout } from './FoldableColumnLayout';

describe('FoldableColumnLayout', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <FoldableColumnLayout sidebarContent={<span>sidebar content</span>}>
        <span>main content</span>
      </FoldableColumnLayout>,
    );
    expect(container).toMatchSnapshot();
  });
});
