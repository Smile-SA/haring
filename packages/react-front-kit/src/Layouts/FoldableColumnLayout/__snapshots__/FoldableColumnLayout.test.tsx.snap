// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoldableColumnLayout matches snapshot 1`] = `
<div>
  <div
    class=" mantine-1echu65"
    color="primary"
  >
    <div
      class="mantine-Grid-root mantine-18hdle4"
    >
      <div
        class="mantine-Grid-col mantine-psscdt"
      >
        <div
          class="mantine-Switch-root mantine-Switch-root mantine-1knyxmk"
        >
          <div
            class="mantine-hqx0d1 mantine-Switch-body"
          >
            <input
              checked=""
              class="mantine-19htmtn mantine-Switch-input"
              id="sidebar-toggle"
              type="checkbox"
            />
            <label
              class="mantine-1i8ydp7 mantine-Switch-track"
              for="sidebar-toggle"
            >
              <div
                class="mantine-1nqstp3 mantine-Switch-thumb"
              />
              <div
                class="mantine-r9fta7 mantine-Switch-trackLabel"
              />
            </label>
            <div
              class="mantine-g88pth mantine-Switch-labelWrapper"
            >
              <label
                class="mantine-14fko7b mantine-Switch-label"
                for="sidebar-toggle"
              >
                <div
                  class="mantine-Text-root mantine-1o8aepr"
                >
                  Display sidebar
                </div>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mantine-16nq4t3 "
  >
    <button
      class="mantine-UnstyledButton-root mantine-Button-root mantine-s27q6v"
      data-button="true"
      type="button"
    >
      <div
        class="mantine-1wpc1xj mantine-Button-inner"
      >
        <span
          class="mantine-1ryt1ht mantine-Button-label"
        >
          Display sidebar
        </span>
        <span
          class="mantine-Button-icon mantine-Button-rightIcon mantine-467wo8"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M213.66,165.66a8,8,0,0,1-11.32,0L128,91.31,53.66,165.66a8,8,0,0,1-11.32-11.32l80-80a8,8,0,0,1,11.32,0l80,80A8,8,0,0,1,213.66,165.66Z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    class="mantine-Container-root mantine-1tcnuk"
  >
    <div
      class="mantine-Grid-root mantine-dgp6xz"
    >
      <div
        aria-hidden="false"
        class="mantine-Grid-col mantine-1psi0ra"
      >
        <span>
          sidebar content
        </span>
      </div>
      <div
        class="mantine-Grid-col mantine-18kcs1a"
      >
        <span>
          main content
        </span>
      </div>
    </div>
  </div>
</div>
`;
